provider/core.rb

```diff
@@ -461,9 +461,32 @@ def emit_method(name, args, code, file_name, opts = {})
     end
 
     def method_decl(name, args)
```

Comment:
The fact that this happens to work for ruby and python is awesome - go ahead and document that, though?  For instance, if anyone ever needed to add kwargs they'd be annoyed to find out that they broke (puppet & chef) | (ansible).